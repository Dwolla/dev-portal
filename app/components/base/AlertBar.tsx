import React, { useState, useEffect } from "react";
import styled from "@emotion/styled";
import infoIcon from "../../../assets/images/component-icons/info-icon.svg";
import warningIcon from "../../../assets/images/component-icons/warning-icon.svg";
import announcementIcon from "../../../assets/images/component-icons/announcement-icon.svg";
import { ReactComponent as CloseIcon } from "../../../assets/images/component-icons/close.svg";

import {
  GREY_1,
  GREY_2,
  API_STATUS_BAR_YELLOW,
  PARAGRAPH_TEXT,
  ANNOUNCEMENT_BACKGROUND_BLUE,
  ANNOUNCEMENT_ICON_BLUE,
  ANNOUNCEMENT_TEXT_BLUE,
} from "../colors";
import { ROBOTO } from "../typography";

const StyledAlertBar = styled.div`
  display: flex;
  width: 100%;
  padding: 15px;
  box-sizing: border-box;
  &.info {
    border: 1px solid ${GREY_2};
    border-radius: 5px;
    background-color: ${GREY_1};
  }
  &.warning {
    border: 1px solid ${API_STATUS_BAR_YELLOW};
    border-radius: 5px;
  }
  &.announcement {
    background-color: ${ANNOUNCEMENT_BACKGROUND_BLUE};
    border-radius: 5px;
    * {
      justify-content: space-around;
      color: ${ANNOUNCEMENT_TEXT_BLUE};
      font-style: normal;
      font-weight: 400;
      font-size: 14px;
      line-height: 143%;
      letter-spacing: 0.17px;
    }
  }
`;

const ImageWrapper = styled.div`
  flex-shrink: 0;
  height: 24px;
  width: 24px;
`;

const TextWrapper = styled.div`
  flex-grow: 2;
  padding-left: 13px;
  color: ${PARAGRAPH_TEXT};
  font-family: ${ROBOTO};
  font-size: 15px;
  font-weight: 500;
  line-height: 25px;
`;

const StyledCloseIcon = styled.div`
  margin-left: 10px;
  cursor: pointer;
  svg {
    * {
      fill: ${ANNOUNCEMENT_TEXT_BLUE};
    }
  }

  :hover,
  :focus {
    svg {
      * {
        fill: ${ANNOUNCEMENT_ICON_BLUE};
      }
    }
  }

  :focus {
    outline: none;
  }
`;

type Props = {
  variation: "warning" | "info" | "announcement";
  isClosable?: boolean;
  children: any;
};

function AlertBar({ variation, isClosable, children }: Props) {
  const [showAlert, setShowAlert] = useState(false);

  /* On component mount, it will check the variation of the component. If it is an info or warning,
  then showalert is set to TRUE, and the component is displayed. If it is an announcement,
  then it will check if cookie exists. If it does, the AlertBar is not displayed. */
  useEffect(() => {
    if (variation === "info" || variation === "warning") {
      setShowAlert(true);
    } else if (document.cookie.indexOf("dwolla_announcement_cookie") === -1) {
      setShowAlert(true);
    } else {
      setShowAlert(false);
    }
  });

  /* Hides Alertbar and sets "dwolla_announcement_cookie" cookie when the announcement alertbar
  is closed by mouseclick of keyboard enter key.
  max-age=31536000 means the cookie will expire in a year.
  Change max-age value as needed. */
  const hideAlert = () => {
    setShowAlert(false);
    document.cookie = "dwolla_announcement_cookie=1; max-age=31536000";
  };
  function onKeydown(e) {
    if (e.key === "Enter") hideAlert();
  }

  // Set icon based on variation selected
  let icon;
  switch (variation) {
    case "warning":
      icon = warningIcon;
      break;
    case "info":
      icon = infoIcon;
      break;
    case "announcement":
      icon = announcementIcon;
      break;
    default:
      break;
  }

  return (
    // Render component as long as close button isn't clicked
    // eslint-disable-next-line react/jsx-no-useless-fragment
    <>
      {showAlert && (
        <StyledAlertBar className={variation}>
          <ImageWrapper>
            <img src={icon} alt={variation} />
          </ImageWrapper>
          <TextWrapper>{children}</TextWrapper>
          {isClosable && (
            <StyledCloseIcon
              tabIndex={0}
              onClick={hideAlert}
              onKeyDown={(keyPress) => onKeydown(keyPress)}
            >
              <CloseIcon width={16} />
            </StyledCloseIcon>
          )}
        </StyledAlertBar>
      )}
    </>
  );
}

export default AlertBar;
