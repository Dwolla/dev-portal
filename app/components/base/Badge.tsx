import React from "react";
import styled from "@emotion/styled";
import {
  ORANGE_PRIMARY,
  PURPLE_023,
  PURPLE_087,
  WHITE_PRIMARY,
} from "../colors";
import { ROBOTO } from "../typography";

const StyledBadge = styled.div`
  color: ${PURPLE_087};
  font-family: ${ROBOTO};
  font-size: 13px;
  font-weight: 400;
  font-style: normal;
  letter-spacing: 0.16px;
  line-height: 18px;
  text-align: center;
  box-sizing: border-box;
  border: 1.3px solid ${PURPLE_023};
  border-radius: 15px;
  padding: 5px 7px;
  &.orange {
    background-color: ${ORANGE_PRIMARY};
    border-color: ${ORANGE_PRIMARY};
    border-radius: 15px;
    color: ${WHITE_PRIMARY};
    font-size: 13px;
    letter-spacing: 0;
    height: 22px;
    padding: 2px 8px;
  }
`;

type Props = { text: string; variant?: "default" | "orange" };

function Badge({ text, variant }: Props) {
  return <StyledBadge className={variant}>{text}</StyledBadge>;
}
export default Badge;
