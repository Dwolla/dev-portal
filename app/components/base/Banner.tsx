import styled from "@emotion/styled";
import classnames from "../../modules/classnames";
import Button from "./Button";
import { PURPLE_DARK, WHITE_PRIMARY, GREY_4 } from "../colors";
import { POPPINS, ROBOTO } from "../typography";
import { breakDown } from "../breakpoints";

// Styled Banner component
const StyledBanner = styled.div`
  background-color: ${PURPLE_DARK};
  padding: 40px;
  display: flex;
  flex-direction: column;
  align-items: center;
  position: relative;
  width: 100%;

  &.overlapped {
    padding-bottom: 75px;
  }

  &.filters {
    padding-bottom: 70px;
  }
`;

const BannerContentWrap = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;

  > *:last-child {
    margin-bottom: 0;
  }
`;

const StyledIcon = styled.img`
  height: 46px;
  width: 42px;
  margin-bottom: 20px;
`;

const StyledTopic = styled.div`
  max-width: 749px;
  color: ${WHITE_PRIMARY};
  font-family: ${POPPINS};
  font-size: 42px;
  font-weight: 200;
  line-height: 52px;
  text-align: center;
  margin-bottom: 20px;

  @media (${breakDown("sm")}) {
    font-size: 30px;
    line-height: 46px;
  }

  @media (${breakDown("xs")}) {
    font-size: 22px;
    line-height: 33px;
  }
`;

const StyledDescription = styled.div`
  color: ${GREY_4};
  font-family: ${ROBOTO};
  font-size: 19px;
  font-weight: 300;
  line-height: 26px;
  text-align: center;
  margin-bottom: 30px;

  @media (${breakDown("sm")}) {
    font-size: 16px;
    line-height: 20px;
  }
`;

const StyledFilter = styled.div`
  text-transform: uppercase;
  position: absolute;
  bottom: 0;
`;

// Button props
export type ButtonProps = {
  text: string;
  link: {
    href: string;
    external: boolean;
  };
};

// Prop Types
type Props = {
  icon?: string;
  topic: string;
  description: string;
  button?: ButtonProps;
  filterTabs?: JSX.Element;
  variant?: "default" | "overlapped";
};

// Banner component
function Banner({
  icon,
  topic,
  description,
  button,
  filterTabs,
  variant = "default",
}: Props) {
  return (
    <StyledBanner className={classnames({ filters: filterTabs }, variant)}>
      <BannerContentWrap>
        {icon && <StyledIcon src={icon} alt="" />}
        <StyledTopic>{topic}</StyledTopic>
        <StyledDescription>{description}</StyledDescription>
        {button && <Button {...button} size="large" variant="primary" />}
      </BannerContentWrap>
      {filterTabs && <StyledFilter>{filterTabs}</StyledFilter>}
    </StyledBanner>
  );
}

export default Banner;
