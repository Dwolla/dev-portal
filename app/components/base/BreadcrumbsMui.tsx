import * as React from "react";
import Breadcrumbs from "@mui/material/Breadcrumbs";
import Link from "@mui/material/Link";
import Typography from "@mui/material/Typography";
import { useRouter } from "next/router";
import { FrontMatter } from "../../../layouts/types";

// Custom word mapping for acronyms
const customTitleMapping = {
  "api-reference": "API Reference",
  "sdks-tools": "SDKs & Tools",
  sdks: "SDKs",
  kba: "KBA",
  // Add more custom words as needed
};

// Utility function to capitalize the words of a string
function capitalizeWords(str: string): string {
  return str.replace(/-/g, " ").replace(/\b\w/g, (c) => c.toUpperCase());
}

// Utility function to generate the breadcrumb list
function generateBreadcrumbs(
  asPathNestedRoutes: string[],
  frontMatter: FrontMatter
) {
  const breadcrumbs = [];
  let currentPath = "";

  for (let i = 0; i < asPathNestedRoutes.length; i += 1) {
    const subpath = asPathNestedRoutes[i];
    currentPath += `/${subpath}`;

    // Skip the "Docs" segment if it's the first segment and starts with "docs"
    if (i === 0 && subpath.startsWith("docs")) {
      // eslint-disable-next-line no-continue
      continue; // Skip the "Docs" segment
    }

    // Use customTitleMapping if available, or fallback to the original segment.
    const text = customTitleMapping[subpath] || subpath;

    if (i === asPathNestedRoutes.length - 1) {
      // If it's the last segment, use the title from frontMatter if available,
      // or fallback to the default text.
      breadcrumbs.push({
        href: currentPath,
        text: frontMatter?.["group"]?.title || frontMatter?.["title"] || text, // eslint-disable-line dot-notation
      });
    } else {
      // For non-last segments, capitalize the text and add it to the breadcrumbs.
      breadcrumbs.push({
        href: currentPath,
        text: capitalizeWords(text),
      });
    }
  }

  // Add "Home" breadcrumb to the list
  return [{ href: "/", text: "Home" }, ...breadcrumbs];
}

type Props = {
  frontMatter?: FrontMatter;
};

type CrumbProps = {
  text: string;
  href: string;
  last: boolean;
};

function Crumb({ text, href, last = false }: CrumbProps) {
  if (last) {
    return <Typography color="text.primary">{text}</Typography>;
  }

  return (
    <Link underline="hover" color="inherit" href={href}>
      {text}
    </Link>
  );
}

export default function BreadcrumbsMui({ frontMatter }: Props) {
  const router = useRouter();

  // Split the router's asPath into segments and remove empty segments
  const asPathWithoutQuery = router.asPath.split("?")[0];
  const asPathNestedRoutes = asPathWithoutQuery
    .split("/")
    .filter((v) => v.length > 0);

  // Generate breadcrumbs based on the asPath segments and frontMatter
  const breadcrumbs = generateBreadcrumbs(asPathNestedRoutes, frontMatter);

  return (
    <Breadcrumbs aria-label="breadcrumb">
      {breadcrumbs.map((crumb, idx) => (
        <Crumb
          {...crumb}
          key={crumb.text}
          last={idx === breadcrumbs.length - 1}
        />
      ))}
    </Breadcrumbs>
  );
}
