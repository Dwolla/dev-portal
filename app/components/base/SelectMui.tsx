import React, { ComponentPropsWithRef, useId } from "react";
import { InputLabel, MenuItem, Select, SxProps } from "@mui/material";
import { MUI_DROPDOWN_FOCUS, WHITE_PRIMARY } from "../colors";
import ConditionalWrapper from "../util/ConditionalWrapper";

/**
 * Defines a `<select>` option that was chosen, returning both its label and value.
 */
export interface SelectMuiOption {
  label: string;
  value: SelectMuiValue;
  icon?: React.FunctionComponent<React.SVGProps<SVGSVGElement>>;
  // eslint-disable-next-line no-unused-vars
  href?: (selectedProduct?: SelectMuiValue) => string;
}

/**
 * Defines all properties that can be specified with the {@link SelectMui} component.
 */
export interface SelectMuiProps {
  color?: SelectMuiColor;
  label: string;
  // eslint-disable-next-line no-unused-vars
  onChange: (value: SelectMuiOption) => MaybePromise<void>;
  options: Array<SelectMuiOption>;
  value: SelectMuiOption | SelectMuiValue;
}

/**
 * Defines all acceptable color values. If a new color is added to this type definition,
 * it must have a mapping associated with it in the {@link colorMap} variable below.
 */
type SelectMuiColor = "white";

/**
 * Defines the indexed type from {@link SelectMuiColor} to all its respective hex values.
 */
type SelectMuiColorMap = {
  // eslint-disable-next-line no-unused-vars
  [_ in SelectMuiColor]: { color: string; focusedColor: string };
};

/**
 * Defines the value that is passed to the embedded MUI {@link Select} component. Defaults to
 * any value that React will accept for an `<li>` element.
 */
type SelectMuiValue = ComponentPropsWithRef<"li">["value"];

/**
 * Map all {@link SelectMuiColor}s to their respective CSS HEX values.
 */
const colorMap: SelectMuiColorMap = {
  white: {
    color: WHITE_PRIMARY,
    focusedColor: MUI_DROPDOWN_FOCUS,
  },
};

/**
 * Get the MUI {@link SxProps} modifications that are displayed with the {@link InputLabel}
 * component.
 *
 * This includes the text color, both in a default and in a focused state.
 */
function getInputColor(color?: SelectMuiColor): SxProps | undefined {
  const { color: muiColor, focusedColor: muiFocusedColor } =
    colorMap[color] || {};

  return (
    muiColor &&
    muiFocusedColor && {
      color: muiColor,
      "&.Mui-focused": { color: muiFocusedColor },
    }
  );
}

/**
 * Get the MUI {@link SxProps} modifications that are displayed with the {@link Select} component.
 *
 * This includes the border color (both in a default or focused state), the hover border color and
 * thickness, and the color of both the text and the embedded svg element.
 */
function getSelectColor(color?: SelectMuiColor): SxProps | undefined {
  const { color: muiColor, focusedColor: muiFocusedColor } =
    colorMap[color] || {};

  return (
    muiColor &&
    muiFocusedColor && {
      color: muiColor,
      "&.MuiOutlinedInput-root": {
        "& fieldset": { borderColor: muiColor },
        "&.Mui-focused fieldset": { borderColor: muiFocusedColor },
        "&:hover fieldset": { borderColor: muiColor, borderWidth: "2px" },
        "& svg": { color },
      },
    }
  );
}

/**
 * Given a value (which can be of type {@link SelectMuiOption} or {@link SelectMuiValue}),
 * get if it is of type {@link SelectMuiOption}.
 */
function isOption(
  obj: SelectMuiOption | SelectMuiValue
): obj is SelectMuiOption {
  return typeof obj === "object" && "label" in obj && "value" in obj;
}

export default function SelectMui({
  color,
  label,
  onChange,
  options,
  value,
}: SelectMuiProps): JSX.Element {
  const id = useId();

  return (
    <>
      <InputLabel id={id} sx={getInputColor(color)}>
        {label}
      </InputLabel>
      <Select<SelectMuiValue>
        label={label}
        labelId={id}
        onChange={({ target: { value: selectedValue } }) => {
          if (onChange) {
            onChange(options.find((obj) => obj.value === selectedValue));
          }
        }}
        sx={getSelectColor(color)}
        value={isOption(value) ? value.value : value}
        variant="outlined"
      >
        {options.map(
          ({ label: itemLabel, value: itemValue, icon: ItemIcon }) => (
            <MenuItem key={useId()} value={itemValue}>
              {/* ConditionalWrapper is used to display the ItemIcon alongside the itemLabel
              if one exists */}
              <ConditionalWrapper
                condition={!!ItemIcon}
                wrapper={(children) => (
                  <div
                    style={{
                      display: "flex",
                      alignItems: "center",
                      justifyItems: "space-between",
                    }}
                  >
                    <ItemIcon width={24} />
                    <div style={{ marginLeft: "0.5em", padding: "0.3em 0" }}>
                      {children}
                    </div>
                  </div>
                )}
              >
                {itemLabel}
              </ConditionalWrapper>
            </MenuItem>
          )
        )}
      </Select>
    </>
  );
}
