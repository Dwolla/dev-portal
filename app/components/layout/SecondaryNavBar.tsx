import React, { useEffect } from "react";
import { useRouter } from "next/router";
import styled from "@emotion/styled";
import NextLink from "next/link";
import { FormControl } from "@mui/material";
import AppBar from "@mui/material/AppBar";
import Box from "@mui/material/Box";
import Toolbar from "@mui/material/Toolbar";
import Container from "@mui/material/Container";
import Button from "@mui/material/Button";
import SelectMui, { SelectMuiOption } from "../base/SelectMui";
import { PURPLE_023, PURPLE_075, PURPLE_100, WHITE_PRIMARY } from "../colors";
import { ROBOTO } from "../typography";
import { breakDown } from "../breakpoints";

export const NAV_BAR_HEIGHT = 70;

const ProductSelectorWrapper = styled.div`
  align-self: center;
  margin: 0 10px 0 0;
  padding: 0 20px 0 0;
  width: 400px; // Value of LEFT_SIDEBAR_WIDTH - 20px from ../../../app/components/layout/Layout.tsx

  @media (${breakDown("md")}) {
    width: 300px;
    margin: unset;
    padding: unset;
  }
`;

const NavBarWrapper = styled.div`
  display: flex;
  width: calc(100vw - 420px);
  padding-right: 2em;
`;

export type SecondaryNavBarProps = {
  navItems: SelectMuiOption[];
  selectedSecondaryNavItem: SelectMuiOption;
  setSelectedSecondaryNavItem: Function;
  productOptions: SelectMuiOption[];
  selectedProduct: SelectMuiOption;
  setSelectedProduct: Function;
};

export default function SecondaryNavBar({
  navItems,
  selectedSecondaryNavItem,
  setSelectedSecondaryNavItem,
  productOptions,
  selectedProduct,
  setSelectedProduct,
}: SecondaryNavBarProps) {
  const router = useRouter();

  useEffect(() => {
    // Find the corresponding nav item based on the current path
    const currentNavItem = navItems.find((item, index) =>
      index === 0
        ? undefined
        : router.asPath.startsWith(item.href(String(selectedProduct?.value)))
    );

    // Update selectedSecondaryNavItem based on the current path
    setSelectedSecondaryNavItem(currentNavItem || navItems[0]);
  }, [router.asPath, selectedProduct]);

  return (
    <AppBar
      position="sticky"
      sx={{
        bgcolor: WHITE_PRIMARY,
        boxShadow: "none",
        borderBottom: `1px solid ${PURPLE_023}`,
        minHeight: NAV_BAR_HEIGHT,
        display: { xs: "none", sm: "none", md: "none", lg: "flex" },
      }}
    >
      <Container maxWidth={false}>
        <Toolbar disableGutters sx={{ justifyContent: "space-around" }}>
          <Box
            sx={{
              flexGrow: 1,
              display: { xs: "none", md: "flex" },
              justifyContent: "space-between",
              minHeight: NAV_BAR_HEIGHT,
            }}
          >
            <ProductSelectorWrapper>
              <FormControl sx={{ m: 1, width: "100%" }} size="small">
                <SelectMui
                  label="Select Product"
                  onChange={(newProduct) => {
                    setSelectedProduct(newProduct);
                    router.push(
                      selectedSecondaryNavItem.href(newProduct.value)
                    );
                  }}
                  options={productOptions}
                  value={selectedProduct || ""}
                />
              </FormControl>
            </ProductSelectorWrapper>

            <NavBarWrapper>
              {navItems.map((item) => (
                <NextLink
                  key={item.label}
                  href={item.href(String(selectedProduct?.value))} // Generate href based on
                  // selectedProduct
                >
                  <Button
                    key={item.label}
                    onClick={() => {
                      setSelectedSecondaryNavItem(item);
                    }}
                    sx={{
                      font: ROBOTO,
                      display: "block",
                      width: "11rem",
                      marginBottom: "unset",
                      borderRadius: "0px",
                      color:
                        item?.value === selectedSecondaryNavItem?.value
                          ? PURPLE_100
                          : PURPLE_075,
                      borderBottom:
                        item?.value === selectedSecondaryNavItem?.value
                          ? `2px solid ${PURPLE_100}`
                          : `2px solid transparent`,
                      "&:hover": {
                        color: PURPLE_100,
                        borderBottom: `2px solid ${PURPLE_100}`,
                      },
                      "&:focus": {
                        color: PURPLE_100,
                        borderBottom: `2px solid ${PURPLE_100}`,
                      },
                    }}
                  >
                    {item.label}
                  </Button>
                </NextLink>
              ))}
            </NavBarWrapper>
          </Box>
        </Toolbar>
      </Container>
    </AppBar>
  );
}
