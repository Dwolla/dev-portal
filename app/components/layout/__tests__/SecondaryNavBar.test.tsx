import React from "react";
import renderer from "react-test-renderer";
import SecondaryNavBar from "../SecondaryNavBar";
import { SelectMuiOption } from "../../base/SelectMui";
import { ReactComponent as DwollaConnectColorIcon } from "../../../../assets/images/product-icons-and-heroes/dwolla-connect-icon-48x48.svg";
import { ReactComponent as DwollaBalanceColorIcon } from "../../../../assets/images/product-icons-and-heroes/dwolla-balance-icon-48x48.svg";

// Mock the useRouter hook
jest.mock("next/router", () => ({
  ...jest.requireActual("next/router"),
  useRouter: () => ({
    asPath: "/mocked-path", // Provide a mocked asPath value for testing
    push: jest.fn(),
  }),
}));

const PRODUCT_OPTIONS: SelectMuiOption[] = [
  { value: "connect", label: "Dwolla Connect", icon: DwollaConnectColorIcon },
  { value: "balance", label: "Dwolla Balance", icon: DwollaBalanceColorIcon },
];

const NAV_ITEMS: SelectMuiOption[] = [
  {
    value: "guidesAndConcepts",
    label: "Guides and Concepts",
    href: (selectedProduct) => `/docs/${selectedProduct}`, // Dynamic href, changes based on selectedProduct Context
  },
  {
    value: "apiReference",
    label: "API Reference",
    href: (selectedProduct) => `/docs/${selectedProduct}/api-reference`, // Dynamic href, changes based on selectedProduct Context
  },
  {
    value: "codeSamples",
    label: "Code Samples",
    href: () => "/code-samples", // Static href, unchanged
  },
  {
    value: "dropIns",
    label: "Drop Ins",
    href: () => "/docs/drop-in-components", // Static href, unchanged
  },
  {
    value: "sdks",
    label: "SDKs",
    href: () => "/sdks-tools", // Static href, unchanged
  },
  {
    value: "changelog",
    label: "Changelog",
    href: () => "/changelog", // Static href, unchanged
  },
];

const mockSelectedSecondaryNavItem = NAV_ITEMS[0];
const mockSelectedProduct = PRODUCT_OPTIONS[0];

const mockSetSelectedSecondaryNavItem = jest.fn();
const mockSetSelectedProduct = jest.fn();

test("SecondaryNavBar snapshot", () => {
  const tree = renderer
    .create(
      <SecondaryNavBar
        navItems={NAV_ITEMS}
        selectedSecondaryNavItem={mockSelectedSecondaryNavItem}
        setSelectedSecondaryNavItem={mockSetSelectedSecondaryNavItem}
        productOptions={PRODUCT_OPTIONS}
        selectedProduct={mockSelectedProduct}
        setSelectedProduct={mockSetSelectedProduct}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
