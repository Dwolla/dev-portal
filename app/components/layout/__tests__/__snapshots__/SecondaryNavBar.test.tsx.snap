// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondaryNavBar snapshot 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionSticky css-pjew9n-MuiPaper-root-MuiAppBar-root"
>
  <div
    className="MuiContainer-root css-ur2jdm-MuiContainer-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular css-1mojz2a-MuiToolbar-root"
    >
      <div
        className="MuiBox-root css-xh3wsv"
      >
        <div
          className="css-7ui14r-ProductSelectorWrapper emsji1v1"
        >
          <div
            className="MuiFormControl-root css-78me8v-MuiFormControl-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={true}
              id="select-mui-0"
            >
              Select Product
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
              onClick={[Function]}
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="select-mui-0"
                className="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-jedpe8-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                role="button"
                tabIndex={0}
              >
                <div
                  style={
                    {
                      "alignItems": "center",
                      "display": "flex",
                      "justifyItems": "space-between",
                    }
                  }
                >
                  <svg
                    width={24}
                  />
                  <div
                    style={
                      {
                        "marginLeft": "0.5em",
                        "padding": "0.3em 0",
                      }
                    }
                  >
                    Dwolla Connect
                  </div>
                </div>
              </div>
              <input
                aria-hidden={true}
                aria-invalid={false}
                className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                disabled={false}
                onAnimationStart={[Function]}
                onChange={[Function]}
                required={false}
                tabIndex={-1}
                value="connect"
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-14lo706"
                >
                  <span>
                    Select Product
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="css-et77cy-NavBarWrapper emsji1v0"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1elaevu-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            Product Overview
            <span
              className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1v1wdku-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            API Reference
            <span
              className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1v1wdku-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            Code Samples
            <span
              className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1v1wdku-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            Drop Ins
            <span
              className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1v1wdku-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            SDKs
            <span
              className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1v1wdku-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            Changelog
            <span
              className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</header>
`;
